### PR TITLE
Use URL constructor in the koa2 redirect callback

### PR DESCRIPTION
**Pull request by abdonrd**
Build the URL using the native `URL` & `URLSearchParams` JavaScript constructors instead of a library to parse the params and a manual string concatenation.

### CODE REVIEW

**Comment by simov** [comment]
Hi @abdonrd,

What's the benefit in using this approach?

**Comment by abdonrd** [comment]
@simov build the URL using the native `URL` & `URLSearchParams` JavaScript constructors instead of a library to parse the params and a manual string concatenation.

**Comment by simov** [comment]
First of all - that's a great idea!

However, there are a few problems not necessarily related to this PR. 

All middlewares (consumers) in this project are basically identical, meaning that this change should be applied in all of them, just to keep them consistent and easier to maintain. But the `URL` and the `URLSearchParams` classes are only available in Node 10 with latest V8. That means that the tests are going to fail without a polyfill, and also this polyfill will be required on versions of Node below 10. So in this case I'm trading the `qs` dependency for another one.

Why the tests are currently not catching this? Well that's another problem related to supporting older versions of Node. Currently the Koa2 middleware is not being executed in Travis CI, exactly because it will fail on older versions, and because it's intentionally not being transpiled.

In summary: migrating to `URL` and `URLSearchParams` and dropping the `qs` dependency altogether is a great idea, but it can't happen right now. And also I have a bit longer support window for Node versions. In the meantime I really don't think that using `qs` and concatenating strings is inferior in any way. In fact it's been battle tested for years.

I hope you understand, and thanks for your interest in this project :+1:

**Comment by abdonrd** [comment]
Sure, I understand! No problem!

I just saw that `URL` & `URLSearchParams` works from [Node.js v6](https://nodejs.org/docs/latest-v6.x/api/url.html). What is the minimum version that `grant` must support? I thought about the [current & active LTS versions](https://github.com/nodejs/Release#nodejs-release-working-group) and later.

**Comment by simov** [comment]
Yes, you are correct. I got misled by the fact that these two classes are global in the browser, which makes sense, but they are in fact global in [node 10](https://nodejs.org/dist/latest-v10.x/docs/api/globals.html) too. So when I quickly checked in older version they were not available as globals.

As for the LTS versions, I generally try to support Node versions a few more months after their EOL, so that people can actually have the time to migrate without me breaking anything.

That being said I can definitely drop support for Node 4 at least, but then again, if I'm going to migrate to `URL` I probably would like to drop the `qs` dependency altogether, again for consistency reasons. I'm adding this to my TODO list. Your feedback was definitely helpful!

**Comment by abdonrd** [comment]
It makes all the sense! Thanks @simov, happy to have helped! 😄
